src/glmfit.jl

```diff
@@ -850,12 +850,11 @@ function momentmatrix(m::GeneralizedLinearModel)
 end
 
 function varstruct(x::GeneralizedLinearModel)
-    wrkwt = working_weights(x)
+    wrkwts = working_weights(x)
     wts = weights(x)
-    wrkwts = wts isa ProbabilityWeights ? wrkwt .* (nobs(x) ./ sum(wts)) : wrkwt
+    wts isa ProbabilityWeights && (wrkwts .*= nobs(x) / sum(wts))
```

Comment:
Hmm, looking at this again I wonder whether my suggestion may have been too ambitious. I'm not sure it's OK to overwrite working weights. If that doesn't work we'll have to do something like this instead:
```suggestion
    wts isa ProbabilityWeights && (wrkwts = wrkwts .* (nobs(x) / sum(wts)))
```

Reply:
Or actually put more code in the `if` to still avoid an allocation:
```julia
if wts isa ProbabilityWeights
    return wrkwts .* wrkres .* (nobs(x) / sum(wts)
else
    return wrkwts .* wrkres
end
```